test/test_distributions.py

```diff
@@ -1316,8 +1338,11 @@ def test_one_hot_categorical_shape(self):
         self.assertEqual(dist.log_prob(dist.enumerate_support()).size(), torch.Size((2, 3)))
         self.assertEqual(dist.log_prob(torch.ones((3, 1, 2))).size(), torch.Size((3, 3)))
 
-    def test_cauchy_shape_scalar_params(self):
-        cauchy = Cauchy(0, 1)
+    def test_cauchy_shape_scalar_params(self, is_half_cauchy=None):
+        if is_half_cauchy:
```

Comment:
It's cleaner to copy the method than to introduce this complex testing logic.

Reply:
code duplication :(

Reply:
cost of abstraction :(